### PR TITLE
stats: speed up row count estimation

### PR DESCRIPTION
**Pull request by lamxTyler**
Before this PR, when binary search on the histogram, we will construct a `Datum` from the chunk for each comparison. In this PR, we directly compare on the chunk without constructing the `Datum`.
Before:
`BenchmarkSelectivity/selectivity-4 50000 24626 ns/op 14232 B/op 195 allocs/op`
After:
`BenchmarkSelectivity/selectivity-4 100000 18757 ns/op 9688 B/op 124 allocs/op`

### CODE REVIEW

**Comment by shenli** [comment]
Almost 100% percent improvement. Good job!

File: statistics/histogram.go [hunk 1]
```diff
@@ -282,113 +281,71 @@ func (hg *Histogram) toString(isIndex bool) string {
...
-	if err != nil {
-		return 0, errors.Trace(err)
-	}
-	eqCount, err := hg.equalRowCount(sc, value)
-	if err != nil {
-		return 0, errors.Trace(err)
-	}
-	return greaterCount + eqCount, nil
+func (hg *Histogram) greaterAndEqRowCount(value types.Datum) float64 {
+	return hg.greaterRowCount(value) + hg.equalRowCount(value)
```
Comment (winoros) [inline statistics/histogram.go]: Why not total-less?

File: statistics/histogram.go [hunk 2]
```diff
@@ -282,113 +281,71 @@ func (hg *Histogram) toString(isIndex bool) string {
...
-func (hg *Histogram) equalRowCount(sc *stmtctx.StatementContext, value types.Datum) (float64, error) {
-	index, match, err := hg.lowerBound(sc, value)
-	if err != nil {
-		return 0, errors.Trace(err)
-	}
-	if index == hg.Len() {
-		return 0, nil
+func (hg *Histogram) equalRowCount(value types.Datum) float64 {
+	index, match := hg.Bounds.LowerBound(0, &value)
+	if index%2 == 1 {
```
Comment (hanfei1991) [inline statistics/histogram.go]: add comment for this line.

File: statistics/histogram.go (hunk 1 above)
Comment (hanfei1991) [inline statistics/histogram.go]: `hg.totalRowCount() - hg.lessRowCount(value)` is better.

**Pull request merged**